### PR TITLE
[DOCS] Update Messaging doc

### PR DESCRIPTION
**Pull request by sijia-w**
### Modifications

- Add intro to message redelivery 
- Update the demo code of `nack` and `ack timeout`
- Modify the content

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [x] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by sijia-w** [comment]
Thanks @gaoran10 for his input and @D-2-Ed for her review comments. Please help review the changes and look forward to your feedback.

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -165,7 +165,7 @@ consumer.acknowledgeCumulative(msg);
 
 ### Negative acknowledgement
 
-When a consumer fails to consume a message and intends to consume it again, this consumer should send a negative acknowledgement to the broker. Then, the broker will redeliver this message to the consumer.
+The negative acknowledgement mechanism allows you to send a notification to the broker indicating the consumer did not process a message.  When a consumer fails to consume a message and needs to re-consume it, the consumer sends a negative acknowledgement (nack) to the broker, triggering the broker to redeliver this message to the consumer.
```
Comment (Anonymitaet) [inline site2/docs/concepts-messaging.md:168]:
```suggestion
The [negative acknowledgement](#negative-acknowledgement) mechanism allows you to send a notification to the broker indicating the consumer did not process a message.  When a consumer fails to consume a message and needs to re-consume it, the consumer sends a negative acknowledgement (nack) to the broker, triggering the broker to redeliver this message to the consumer.
```

File: site2/docs/concepts-messaging.md [hunk 2]
```diff
@@ -202,46 +214,66 @@ consumer.negativeAckRedeliveryBackoff(NegativeAckRedeliveryExponentialBackoff.bu
...
+
+You can configure the acknowledgement timeout mechanism to redeliver the message if it is not acknowledged after `ackTimeout` or to execute a timer task to check the acknowledgement timeout messages during every `ackTimeoutTickTime` period.
 
 > **Note**  
 > - If batching is enabled, all messages in one batch are redelivered to the consumer.  
-> - The negative acknowledgement is preferable over the acknowledgement timeout, since negative acknowledgement controls the redelivery of individual messages more precisely and avoids invalid redeliveries when the message processing time exceeds the acknowledgement timeout.
+> - Compared with acknowledgement timeout, negative acknowledgement is preferred. First, it is difficult to set a timeout value. Second, broker resends messages when the message processing time exceeds the acknowledgement timeout, but these messages might not need to be re-consumed.
```
Comment (Anonymitaet) [inline site2/docs/concepts-messaging.md:223]:
```suggestion
> - Compared with acknowledgement timeout, negative acknowledgement is preferred. First, it is difficult to set a timeout value. Second, a broker resends messages when the message processing time exceeds the acknowledgement timeout, but these messages might not need to be re-consumed.
```

File: site2/docs/concepts-messaging.md [hunk 3]
```diff
@@ -617,6 +649,22 @@ Producer<byte[]> producer = client.newProducer()
                 .create();
 ```
 
+## Message redelivery
+
+Apache Pulsar supports graceful failure handling and ensures critical data is not lost. Software will always have unexpected conditions and at times messages may not be delivered successfully. Therefore, it is important to have a built-in mechanism that handles failure, particularly in asynchronous messaging as highlighted in the following examples.
+
+- Consumers get disconnected from the database or the HTTP server. When this happens, the database is temporarily offline while the consumer is writing the data to it and the external HTTP server that the consumer calls is momentarily unavailable.
```
Comment (Anonymitaet) [inline site2/docs/concepts-messaging.md:656]:
```suggestion
- Consumers get disconnected from the database or the HTTP server. When this happens, the database is temporarily offline while the consumer is writing the data to it and the external HTTP server that the consumer calls are momentarily unavailable.
```

**Review by Anonymitaet — APPROVED** [review]

File: site2/docs/concepts-messaging.md (hunk 3 above)
Comment (sijia-w) [reply site2/docs/concepts-messaging.md:656]: Thanks for all your comments. But here I think the subject is `the external HTTP server`, so no modification is needed.

Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:656]: Oops, my bad.

**Pull request merged**